### PR TITLE
Fix appending to models with existing non-ascii columns

### PR DESCRIPTION
**Pull request by crisptrutski**
This seems to have always been broken, noticed it as part of my clickhouse testing.

### CODE REVIEW

File: test/metabase/upload_test.clj [hunk 1]
```diff
@@ -2391,7 +2426,7 @@
 
 (deftest unique-long-column-names-test
   (let [original ["αbcdεf_αbcdεf"     "αbcdεfg_αbcdεf"   "αbc_2_etc_αbcdεf" "αbc_3_xyz_αbcdεf"]
-        expected [:%CE%B1bcd%  :%_852c229f :%CE%B1bc_2 :%CE%B1bc_3]
+        expected [:%ce%b1bcd%  :%_b59bccce :%ce%b1bc_2 :%ce%b1bc_3]
```
Comment (crisptrutski) [inline test/metabase/upload_test.clj:2429]: The change to the second field might be a blocker 😢 

Might need another way to get around default case differences between databases.

Comment (crisptrutski) [reply test/metabase/upload_test.clj:2429]: We decided its OK to break appends to existing files with duplicated non-ascii headers. Appending didn't work without this change, and maintaining backwards compatibility with those existing columns would introduce a lot of complexity.

File: src/metabase/upload.clj [hunk 2]
```diff
@@ -756,8 +757,7 @@
               [header & rows]    (cond-> (parse reader)
                                    auto-pk?
                                    without-auto-pk-columns)
-              normed-name->field (m/index-by #(normalize-column-name driver (:name %))
-                                             (t2/select :model/Field :table_id (:id table) :active true))
+              normed-name->field (m/index-by :name (t2/select :model/Field :table_id (:id table) :active true))
```
Comment (crisptrutski) [inline src/metabase/upload.clj:760]: I should rename this to `name->field`...

File: test/metabase/upload_test.clj [hunk 3]
```diff
@@ -1337,13 +1337,22 @@
...
-        insert-col-names (remove #{upload/auto-pk-column-keyword} (keys col->upload-type))
-        col-definitions (#'upload/column-definitions driver col->upload-type)]
+        ->normalized-col  (comp keyword (partial #'upload/normalize-column-name driver) name)
+        name->display     (->> (keys col->upload-type)
+                               (map name)
+                               (remove #{upload/auto-pk-column-name})
+                               ((juxt
+                                 (partial map (comp name ->normalized-col))
+                                 (partial #'upload/derive-display-names driver)))
+                               (apply zipmap))
```
Comment (piranha) [inline test/metabase/upload_test.clj:1352]: woah what is going on here 🤣

Comment (piranha) [reply test/metabase/upload_test.clj:1352]: I mean I parsed it almost fine, but don't you think `(into {} (for ...` would be much more readable than a wild `juxt`?

Comment (crisptrutski) [reply test/metabase/upload_test.clj:1352]: Yeah I let this get a bit out of control, let me try simplify it. "Only test code" 😄

**Review by sansolovyov — APPROVED** [review]
no concerns except a little code complexity bit :)

**Pull request merged**